src/node_util.cc

```diff
@@ -146,6 +148,36 @@ void WatchdogHasPendingSigint(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void CreatePromise(const FunctionCallbackInfo<Value>& args) {
+  Local<Context> context = args.GetIsolate()->GetCurrentContext();
+  auto maybe_resolver = Promise::Resolver::New(context);
+  if (!maybe_resolver.IsEmpty())
+    args.GetReturnValue().Set(maybe_resolver.ToLocalChecked());
+}
```

Comment:
This is internally implemented by basically calling `new Promise()` with no closure argument. However, the great thing is that, since you're using the V8 API, you can still resolve and reject it, even though no closures are allocated (because they won't leak).